anonymizer.py

```diff
@@ -12,214 +10,226 @@ class ServiceRunner(dl.BaseServiceRunner):
-        anon_start = time.time()
-        res = self.anonymize_objects(item, objects_of_interest, model.id, progress, context)
-        print(f"&&&&&& Anonymization time: {time.time() - anon_start}")
-        return res
-
-    def anonymize_annotations(self,
-                              annotations: dl.AnnotationCollection,
-                              progress: dl.Progress,
-                              context: dl.Context) -> dl.Item:
-        logger.info("Starting anonymization based on annotations")
-        _, model_id, labels = self.get_models_and_labels(context)
-        item = annotations[0].item
-        objects_of_interest = dl.AnnotationCollection()
-        annotations_of_interest = dl.AnnotationCollection([ann for ann in annotations if ann.label in labels])
-        for a in annotations_of_interest:
-            objects_of_interest.add(a)
-        anon_start = time.time()
-        res = self.anonymize_objects(item, objects_of_interest, model_id, progress, context)
-        print(f"&&&&&& Anonymization time: {time.time() - anon_start}")
-        return res
+        model_ids = node.metadata['customNodeConfig'].get('model_ids', "")
+        labels = node.metadata['customNodeConfig'].get('labels', "")
+
+        # Process model IDs and labels into lists, ignoring empty values
+        model_ids = [model_id.strip() for model_id in model_ids.split(",") if model_id.strip()]
+        labels = [label.strip() for label in labels.split(",") if label.strip()]
+
+        logger.info("Configuration retrieved:")
+        logger.info(f"Model IDs: {model_ids if model_ids else 'None'}")
+        logger.info(f"Labels: {labels if labels else 'None'}")
+
+        return model_ids, labels
+
+    def anonymize(self, item: dl.Item, context: dl.Context) -> dl.Item:
+        """
+        Anonymize annotations for a given item based on models and labels.
+
+        Args:
+            item (dl.Item): The item to process.
+            context (dl.Context): Context for accessing node configuration.
+
+        Returns:
+            dl.Item: The updated item.
+        """
+        logger.info("Starting anonymization process.")
+        model_ids, labels = self.get_models_and_labels(context)
+
+        # Prepare the base filter for annotations
+        filters = dl.Filters(resource=dl.FiltersResource.ANNOTATION)
+        if labels:  # Add label filter if labels are provided
+            filters.add(dl.KnownFields.LABEL, labels, operator=dl.FiltersOperations.IN)
+
+        # If models are specified, process for each model
+        if model_ids:
+            model_filters = filters
+            model_filters.add(field="metadata.system.model.model_id", values=model_ids, operator=dl.FiltersOperations.IN)
+            logger.info(f"Filtering annotations for model IDs: {model_ids}")
+            objects_of_interest = item.annotations.list(filters=model_filters)
+            item = self.anonymize_objects(item, objects_of_interest, context)
+
+        else:
```

Comment:
why do you need else here?
you always list annotations even if you don't have model ids or labels